### PR TITLE
[WIP] CEDS-3170

### PR DESCRIPTION
**Pull request by MaciejRewera**

### CODE REVIEW

File: test/controllers/declaration/DeclarantDetailsControllerSpec.scala [hunk 1]
```diff
@@ -103,28 +107,47 @@ class DeclarantDetailsControllerSpec extends ControllerSpec {
...
     }
 
-    "return 303 (SEE_OTHER) and redirect to is declarant exporter details page" when {
+    "answer is yes" should {
+
+      onJourney(STANDARD, SUPPLEMENTARY, SIMPLIFIED, OCCASIONAL) { request =>
+        "return 303 (SEE_OTHER) and redirect to Consignment References details page" in {
+
+          withNewCaching(request.cacheModel)
+          val correctForm = JsObject(Map(isEoriKey -> JsString(YesNoAnswers.yes)))
```
Comment (udinobi) [inline test/controllers/declaration/DeclarantDetailsControllerSpec.scala:116]: This could be maybe simplified to `Json.obj(isEoriKey -> YesNoAnswers.yes)` in order to remove the `Map`

File: test/controllers/declaration/DeclarantDetailsControllerSpec.scala [hunk 2]
```diff
@@ -103,28 +107,47 @@ class DeclarantDetailsControllerSpec extends ControllerSpec {
...
+        }
+      }
 
-      "answer is yes" in {
+      onClearance { request =>
+        "return 303 (SEE_OTHER) and redirect to Declarant Exporter page" in {
 
-        val correctForm = JsObject(Map(isEoriKey -> JsString(YesNoAnswers.yes)))
+          withNewCaching(request.cacheModel)
+          val correctForm = JsObject(Map(isEoriKey -> JsString(YesNoAnswers.yes)))
```
Comment (udinobi) [inline test/controllers/declaration/DeclarantDetailsControllerSpec.scala:129]: This could be maybe simplified to `Json.obj(isEoriKey -> YesNoAnswers.yes)` in order to remove the `Map`

File: test/controllers/declaration/DeclarantDetailsControllerSpec.scala [hunk 3]
```diff
@@ -103,28 +107,47 @@ class DeclarantDetailsControllerSpec extends ControllerSpec {
...
-    "return 303 (SEE_OTHER) and redirect to Not Eligible page" when {
+    "answer is no" should {
 
-      "answer is no" in {
+      onEveryDeclarationJourney() { request =>
+        "return 303 (SEE_OTHER) and redirect to Not Eligible page" in {
+          withNewCaching(request.cacheModel)
 
-        val correctForm = JsObject(Map(isEoriKey -> JsString(YesNoAnswers.no)))
+          val correctForm = JsObject(Map(isEoriKey -> JsString(YesNoAnswers.no)))
```
Comment (udinobi) [inline test/controllers/declaration/DeclarantDetailsControllerSpec.scala:145]: likewise

File: test/views/declaration/DeclarantDetailsViewSpec.scala [hunk 4]
```diff
@@ -56,61 +56,72 @@ class DeclarantDetailsViewSpec extends UnitViewSpec with ExportsTestData with Co
...
 
         backButton must containMessage(backCaption)
-        backButton.attr("href") mustBe routes.ConsignmentReferencesController.displayPage().url
+        backButton must haveHref(routes.AdditionalDeclarationTypeController.displayPage().url)
+      }
+
+      "not display 'Save and return' button on page" in {
+
+        val saveButton = createView(form()).getElementById("submit_and_return")
+        saveButton mustBe null
```
Comment (udinobi) [inline test/views/declaration/DeclarantDetailsViewSpec.scala:101]: Why not `Option(createView(form()).getElementById("submit_and_return")) mustBe None` ?

**Review by udinobi — APPROVED** [review]
LGTM, apart a few minor comments

File: test/controllers/declaration/DeclarantDetailsControllerSpec.scala (hunk 2 above)
Comment (MaciejRewera) [reply test/controllers/declaration/DeclarantDetailsControllerSpec.scala:129]: This part was copied from previously existing code. Didn't think about simplifying it, but I like your idea.

File: test/views/declaration/DeclarantDetailsViewSpec.scala (hunk 4 above)
Comment (MaciejRewera) [reply test/views/declaration/DeclarantDetailsViewSpec.scala:101]: I saw in some other place we did it this way. Besides it is a bit more indirect testing - `getElementById` returns `null` if element is not found (as it has Java origins). Not sure about wrapping it in `Option` just to avoid typing `null`. If it was application code then surely this is something to avoid, but in tests I think it is fine.
I'm leaving it as it is.

Comment (udinobi) [reply test/views/declaration/DeclarantDetailsViewSpec.scala:101]: Or `Option(saveButton) mustBe None`

Comment (udinobi) [reply test/views/declaration/DeclarantDetailsViewSpec.scala:101]: :+1:

**Review by udinobi — APPROVED** [review]
